trajectory.py

```diff
@@ -128,6 +121,78 @@ def check_step_is_valid(self, step: np.ndarray, is_bound: bool = False) -> bool:
         bound_check = diff_to_wall < self.bound_zone_thickness
         return bound_check
 
+    def step_mod(self, step: np.ndarray, is_bound: bool = False) -> np.ndarray:
+
+        """
+        When a step is determined to be invalid, this method alters the step
+        to stop at the barrier, both nuclear and bound zone
+
+        Parameters
+        ----------
+        step
+        is_bound
+
+        Returns step altered accordingly
+        -------
+        """
+
+        next_locus_extent = np.linalg.norm(self.position + step) + self.locus_radius
+
+        nuclear_check = self.nuclear_radius > next_locus_extent
+
+        if not nuclear_check:
+
+            ideal_position = self.position + step
+
+            collide_roots = np.roots(
+                [
+                    step[0] ** 2 + step[1] ** 2,
+                    2 * (step[0] + step[1]),
+                    self.position[0] ** 2
+                    + self.position[1] ** 2
+                    - (self.nuclear_radius - 0.01) ** 2,
+                ]
+            )
+
+            if np.abs(collide_roots[0]) < 1:
+                adj_step = step * (np.abs(collide_roots[0]))
+            elif np.abs(collide_roots[1]) < 1:
+                adj_step = step * (np.abs(collide_roots[1]))
+            else:
+                adj_step = step * 0.0001
+
+            return adj_step
+
+        diff_to_wall = self.nuclear_radius - next_locus_extent
+
+        bound_check = diff_to_wall < self.bound_zone_thickness
+
+        if not bound_check:
+
+            # TODO invert the above method for bound zone thickness
+            db_pos = self.position
+
+            ideal_position = self.position + step
+
+            collide_roots = np.roots(
+                [
+                    step[0] ** 2 + step[1] ** 2,
+                    2 * (step[0] + step[1]),
+                    self.position[0] ** 2
+                    + self.position[1] ** 2
+                    - (self.nuclear_radius - self.bound_zone_thickness + 0.0001) ** 2,
+                ]
+            )
+
+            if np.abs(collide_roots[0]) < 1:
+                adj_step = step * (np.abs(collide_roots[0]))
+            elif np.abs(collide_roots[1]) < 1:
+                adj_step = step * (np.abs(collide_roots[1]))
+            else:
+                adj_step = step * 0.0001
+
+            return adj_step
+
```

Comment:
Can split this up

Reply:
Didn't split up but made in the image of unbound